### DISCUSSION TITLE
zip encoding format

### DISCUSSION THREAD

**Discussion by Koyorice** [klauspost/compress#1033] [Ideas/Features]
How to specify the encoding format to use when reading Zip?
If you try to read the Zip file of big5, garbled characters will appear.

**Comment by klauspost** [comment]
There is no reasonable way to detect these, as they rely on assuming the code page used for the encoding.

Use utf-8 when encoding, or plain ascii. 

This behaves similar to the standard library. I would be willing to accept a well made and documented pull request that allows specifying a codepage to use for all files, but you might as well just convert the [(FileHeader).Name](https://pkg.go.dev/github.com/klauspost/compress@v1.17.11/zip#FileHeader) you get returned.

**Comment by Koyorice** [comment]
I don't need to detect the encoding, I just want the ability to specify it explicitly. Since I cannot predict the encoding type used by the user, this causes issues when using `fs.WalkDir`.

**Reply by klauspost** [reply]
You are welcome to send a PR. If it is clean and well documented I will include it. But I don't have any significant bandwidth to throw at it.

**Reply by Koyorice** [reply]
In other words, there is currently no way to set the encoding?

**Reply by klauspost** [reply]
No.

**Reply by Koyorice** [reply]
Okay, thank you!
